Sources/AardvarkReveal/RevealAttachmentGenerator.swift

```diff
@@ -0,0 +1,518 @@
+///                                 ▼
+///   ┌───────────────────────────────────────────────────────────┐
+///   │                                                           │
+///   │                    Capturing Snapshots                    │
+///   │                                                           │
+///   │  .─────────────────────────────────────────────────────.  │
+///   │ ( revealAttachmentGeneratorDidCaptureMainScreenSnapshot ) │
+///   │  `─────────────────────────────────────────────────────'  │
+///   └───────────────────────────────────────────────────────────┘
+///                                 │
+///                                 ▼
+///   ┌───────────────────────────────────────────────────────────┐
+///   │                                                           │
+///   │                   Bundling Reveal File                    │
+///   │                                                           │
+///   └───────────────────────────────────────────────────────────┘
+///                                 │
+///    .────────────────────────────┴────────────────────────────.
+///   (   revealAttachmentGeneratorDidFinishBundlingRevealFile    )
+///    `────────────────────────────┬────────────────────────────'
+///                                 ▼
+///                   .───────────────────────────.
+///                  (         completion          )
+///                   `───────────────────────────'
+/// ```
+///
+/// It is important to avoid changing anything while the app state is being captured (during the `Capturing App State`
+/// phase). Once the app state is captured, the generator will begin capturing snapshots of each drawable object (views,
+/// layers, etc.) on the screen. Once the generator has captured a snapshot of the main screen (marked by a call to the
+/// delegate's `revealAttachmentGeneratorDidCaptureMainScreenSnapshot` method), it is okay to display a waiting
+/// indicator **as long as all of the existing views/layers remain in the hierarchy** (for example, you can display a
+/// new window on top of the main window or present a modal view controller, depending on the use case). Once the
+/// generator has completed bundling the Reveal file, user interaction can be restored.
+///
+/// # Security Settings
+///
+/// The generator communicates with the Reveal server over HTTP. By default, the App Transport Security policy requires
+/// the use of a secure connection, which is not available. To enable the generator to connect to the server, add the
+/// following to your app's `Info.plist`:
+///
+/// ```xml
+/// <key>NSAppTransportSecurity</key>
+/// <dict>
+///     <key>NSExceptionDomains</key>
+///     <dict>
+///         <key>localhost</key>
+///         <dict>
+///             <key>NSExceptionAllowsInsecureHTTPLoads</key>
+///             <true/>
+///             <key>NSIncludesSubdomains</key>
+///             <true/>
+///         </dict>
+///     </dict>
+/// </dict>
+/// ```
```

Comment:
I wonder if we can get this to communicate with the Reveal server via the `NetService` directly, rather than HTTP requests, to avoid requiring insecure HTTP requests to `localhost`.

Reply:
@tonyarnold Do you know if communicating via the `NetService` directly would remove the need for this requirement? Or is there an alternate option for communicating with the Reveal server? I don't have much experience with `NetService`, so not sure what options are available.

Reply:
My understanding is that `NetService` is only used for broadcasting availability of services and clients of services. You can't communicate over them directly, so as far as I'm aware, Reveal's `NetService` would be broadcasting details of the HTTP endpoint and the port it is running on to net service browsers. 

I'll have a look when I'm back at my desk later today, and verify exactly what we're doing.

Reply:
Side note: by _golly_ I wish that we offered alternatives here!

We've been a bit hamstrung in terms of some of the nice new technology Apple has added in recent releases - our support for older iOS releases means that we can't use many of them in the server component. I'll need to reassess that now that we're moving our minimum iOS and tvOS versions up to 10.x releases.

Reply:
Ahh, makes sense! Sending the requests over HTTP seems to work great with the exception of this requirement. It would be awesome to get a little bit cleaner alternative, but I definitely don't think this would be a blocker.